mbuild/conversion.py

```diff
@@ -200,44 +220,82 @@ def load_smiles(smiles_or_filename,
     infer_hierarchy : bool, optional, default=True
     ignore_box_warn : bool, optional, default=False
         If True, ignore warning if no box is present.
+    backend : str, optional, default='rdkit'
+        The smiles loading backend, either 'rdkit' or 'pybel'
+    smiles_seed : int, optional, default=0
+        The random number seed to control the position of particles generated
+        from the smiles loader. -1 means no seed is set.
+
 
     Returns
     -------
     compound : mb.Compound
 
     """
-    # Will try to support list of smiles strings in the future
-    pybel = import_('pybel')
-
     # Initialize an mb.Compound if none is provided
     if not compound:
         compound = mb.Compound()
 
-    # First we try treating filename_or_object as a SMILES string
-    try:
-        mymol = pybel.readstring("smi", smiles_or_filename)
-    # Now we treat it as a filename
-    except(OSError, IOError):
-        # For now, we only support reading in a single smiles molecule,
-        # but pybel returns a generator, so we get the first molecule
-        # and warn the user if there is more
-
-        mymol_generator = pybel.readfile("smi", smiles_or_filename)
-        mymol_list = list(mymol_generator)
-        if len(mymol_list) == 1:
-            mymol = mymol_list[0]
+    test_path = Path(smiles_or_filename)
+
+    # Will try to support list of smiles strings in the future
+    if backend == 'rdkit':
+        rdkit = import_('rdkit')
+        from rdkit import Chem
+        from rdkit.Chem import AllChem
+        if test_path.exists():
+            # assuming this is a smi file now
+            mymol = Chem.SmilesMolSupplier(smiles_or_filename)
```

Comment:
Does this function not raise any error for bad input?

Reply:
There is a chance it does if the SMILES is a malformed input. Wouldnt we want that to be handled by RDKit and its error messages? Or would it be better to `try catch` it and then go from there?

Reply:
There's another check just below this line that raises a [`ValueError`](https://github.com/mosdef-hub/mbuild/pull/854/files#diff-1dc110be0119f1cab2f1274217d1e697423688eb07c7d80a13192129b1bb248bR249-R252), which may not be raised if `Chem.SmilesMolSupplier` failed, which instructs user to check the error message from `rdkit`.

Reply:
So, if the `SmilesMolSupplier` cannot create a chemical structure from the SMILES string, it will return [`None` for that object](https://www.rdkit.org/docs/GettingStartedInPython.html#reading-single-molecules).

```python3
>>> m = Chem.MolFromMolFile('data/invalid.mol')
>>> m is None
True
```

Reply:
Okay, That makes sense to me.